clio-server/src/main/scala/org/broadinstitute/clio/server/service/IndexService.scala

```diff
@@ -34,19 +39,46 @@ abstract class IndexService[CI <: ClioIndex](
 
   def upsertMetadata(
     indexKey: clioIndex.KeyType,
-    metadata: clioIndex.MetadataType
-  ): Future[UpsertId] = {
-    val updatedMetadata = metadata.withDocumentStatus(
-      metadata.documentStatus.orElse(Some(DocumentStatus.Normal))
-    )
-
-    persistenceService
-      .upsertMetadata(
-        indexKey,
-        updatedMetadata,
-        documentConverter,
-        elasticsearchIndex
-      )
+    metadata: clioIndex.MetadataType,
+    force: Boolean = false
+  )(implicit materializer: Materializer): Future[UpsertId] = {
+    // query by key to see if the document already exists
+    queryMetadataForKey(indexKey)
+      .runFold[Either[Exception, Json]](
+        // If the query doesn't return a document and the document status is not set default to Normal.
+        Right(
+          metadata.documentStatus
+            .fold(metadata.withDocumentStatus(Some(Normal)))(_ => metadata)
+            .asJson
+        )
+      ) { (queryResult, _) =>
+        // if we are not forcing the we want to run validation
+        if (!force) {
+          queryResult match {
+            case Right(existingMetadata) =>
```

Comment:
Hrm, puzzling over this fold... it doesn't look like it'll guard against > 1 returned metadata. (It'll throw if the > 1 metadatas have different fields, but in the ID fix case the records were an exact match apart from the `entity_id`).